src/java.base/share/classes/java/lang/String.java

```diff
@@ -3599,11 +3599,27 @@ public String repeat(int count) {
      * @param dstBegin  the char index, not offset of byte[]
      * @param coder     the coder of dst[]
      */
-    void getBytes(byte dst[], int dstBegin, byte coder) {
+    void getBytes(byte[] dst, int dstBegin, byte coder) {
+        getBytes(dst, 0, dstBegin, coder, value.length);
+    }
+
+    /**
+     * Copy character bytes from this string into dst starting at dstBegin.
+     * This method doesn't perform any range checking.
+     *
+     * Invoker guarantees: dst is in UTF16 (inflate itself for asb), if two
+     * coders are different, and dst is big enough (range check)
+     *
+     * @param srcPos    the char index, not offset of byte[]
+     * @param dstBegin  the char index to start from
+     * @param coder     the coder of dst[]
+     * @param length    the amount of copied bytes
+     */
+    void getBytes(byte[] dst, int srcPos, int dstBegin, byte coder, int length) {
         if (coder() == coder) {
-            System.arraycopy(value, 0, dst, dstBegin << coder, value.length);
+            System.arraycopy(value, srcPos, dst, dstBegin << coder, length);
```

Comment:
Would it be more consistent and easier to follow if we took the length in chars, not in bytes, and adjusted it here with the coder? JITs should do short work of the need to adjust by calling `length()` in `getBytes(byte[], int, byte)` above.

If we're worried about unexpected performance effects we can separate these two `getBytes` methods to independent methods rather than delegate from one to the other.

Reply:
Done!